api/stocks.go

```diff
@@ -385,3 +364,108 @@ func buildMetricMap(metrics []store.Metric) map[string]store.Metric {
 	}
 	return metricMap
 }
+
+func (s *Server) scoreStock(ctx context.Context, stock *store.Stock) (float64, error) {
+	var totalScore float64
+
+	stockMetrics, err := s.store.FindLatestStockMetrics(ctx, stock.ID)
+	if err != nil {
+		return totalScore, err
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		s.log.Error("Error:", lctx.Err(err))
+	}
+	s.log.Info("Path", lctx.Str("cwd", cwd))
+
+	rules, err := loadScoringRules(s.filePath)
+	if err != nil {
+		return totalScore, err
+	}
+
+	for _, stockMetric := range stockMetrics {
+		rule, exists := rules[stockMetric.MetricName]
+		if !exists {
+			continue
+		}
+
+		totalScore += applyFactors(stockMetric.Value, rule)
+	}
+
+	return totalScore, nil
+}
+
+func getAction(score float64) store.Action {
+	switch {
+	case score >= StrongBuy:
+		return store.StrongBuy
+	case score >= Buy:
+		return store.Buy
+	case score >= Hold:
+		return store.Hold
+	case score >= Sell:
+		return store.Sell
+	default:
+		return store.StrongSell
+	}
+}
+
+func calculateConfidenceLevel(stockMetrics []store.StockMetric) float64 {
+	totalMetrics := len(stockMetrics)
+	if totalMetrics == 0 {
+		return 0
+	}
```

Comment:
`totalMetrics` could be renamed `n` here, or `count`. Those are idiomatically commonly used for counts of items reused throughout a func, and would be shorter